GenericTest.py

```diff
@@ -265,8 +265,8 @@ def validate_schema(self, payload, schema):
         checker = jsonschema.FormatChecker(["ipv4", "ipv6", "uri"])
         return jsonschema.validate(payload, schema, format_checker=checker)
 
-    def do_request(self, method, url, data=None):
-        return TestHelper.do_request(method, url, data)
+    def do_request(self, method, url, data=None, headers=None, auth=None):
+        return TestHelper.do_request(method=method, url=url, data=data, headers=headers, auth=auth)
```

Comment:
Does there need to be a way to pass json to this? Presumably existing uses of this method require that?